share/eds/byzantine/bad_encoding.go

```diff
@@ -132,7 +132,11 @@ func (p *BadEncodingProof) Validate(hdr libhead.Header) error {
 	if int(p.Index) >= len(merkleRowRoots) {
 		return fmt.Errorf("fraud: invalid proof: index out of bounds (%d >= %d)", int(p.Index), len(merkleRowRoots))
 	}
-	if len(merkleRowRoots) != len(p.Shares) {
+	if len(p.Shares) != len(merkleRowRoots) {
```

Comment:
[for reviewers] I switched the order of these so they match the order in the Error statement below. Can revert if requested.